### ISSUE TITLE
Refactor Resource Card Markup

### ISSUE THREAD

**Issue by zdavis** [ManifoldScholar/manifold#2732]
The resource card could use some improvements to make the markup and behavior more semantic. Right now we wrap most of the card in a "section" tag with an onClick handler. We should have a proper a tag here. We also have wonky mouseover/mouseout behavior that leads to flickering hover states. Oddly, there's a div with a "to" attribute with a URL value. What even?

Related to #2727
